napari/layers/labels/labels.py

```diff
@@ -575,6 +585,37 @@ def color(self, color):
 
         self.color_mode = color_mode
 
+    def _validate_colors(self, labels: Iterable[int]):
+        """Check whether any of the given labels will be aliased together.
+
+        See https://github.com/napari/napari/issues/6084 for details.
+        """
+        aliases: Dict[float, List[int]] = defaultdict(list)
+        labels = [k for k in labels if k is not None]
```

Comment:
No need to instantiate the full list/array twice
```suggestion
        labels = (k for k in labels if k is not None)
```